openedx_blockstore_relay/tests/test_transfer_data.py

```diff
@@ -0,0 +1,128 @@
+import mock
+from xml.etree import ElementTree
+
+from django.test import TestCase
+from opaque_keys.edx.keys import UsageKey
+from xblock.core import XBlock
+
+from openedx_blockstore_relay.transfer_data import TransferBlock, transfer_to_blockstore
+from openedx_blockstore_relay.test_utils.compat import StubCompat, StaticContent
+
+
+class TestXBlock(XBlock):
+
+    @property
+    def url_name(self):
+        return self.location.block_id
+
+    def get_children(self):
+        return getattr(self, 'children', [])
+
+    def definition_to_xml(self, resource_fs):
+
+        if self.location.block_type == 'poll':
+            raise NotImplementedError()
+
+        xml_object = ElementTree.Element(self.location.block_type)
+        xml_object.set('display_name', "A test display name")
+        return xml_object
+
+@ddt.ddt
+class TransferBlockTestCase(TestCase):
+    """
+    Tests for TransferBlock.
+    """
+
+    BLOCK_KEY = 'block-v1:edX+DemoX+Demo_Course+type@vertical+block@v1'
+
+    INVALID_COLLECTION_UUID = 'invalid_bundle_uuid'
+    INVALID_BUNDLE_UUID = 'invalid_bundle_uuid'
+
+    COLLECTION_UUID = 'd3e311a8-b3a8-439d-a111-cc6cb99790e8'
+    BUNDLE_UUID = '93fc9c6e-4249-4d57-a63c-b08be9f4fe02'
+
+    def setUp(self):
+        super(TransferBlockTestCase, self).setUp()
+
+        self.block_key = UsageKey.from_string(self.BLOCK_KEY)
+        self.block = TestXBlock(mock.MagicMock(), scope_ids=mock.MagicMock())
+        self.block.location = self.block_key
+
+        child_block_1 = TestXBlock(mock.MagicMock(), scope_ids=mock.MagicMock())
+        child_block_1.location = UsageKey.from_string('block-v1:edX+DemoX+Demo_Course+type@html+block@html1')
+
+        child_block_2 = TestXBlock(mock.MagicMock(), scope_ids=mock.MagicMock())
+        child_block_2.location = UsageKey.from_string('block-v1:edX+DemoX+Demo_Course+type@poll+block@poll1')
+
+        child_block_3 = TestXBlock(mock.MagicMock(), scope_ids=mock.MagicMock())
+        child_block_3.location = UsageKey.from_string('block-v1:edX+DemoX+Demo_Course+type@video+block@video1')
+
+        self.block.children = [child_block_1, child_block_2, child_block_3]
+
+        self.compat = mock.patch('openedx_blockstore_relay.transfer_data.compat', StubCompat({
+            self.block_key: self.block
+        }, assets=[
+            { 'content':StaticContent(''), 'path':'/static/one.jpeg'},
+            { 'content':StaticContent('<script></script>'), 'path':'/static/one.js'},
+        ], video_assets=[
+            {'content':StaticContent('1. Hello welcome to this course.'), 'path': '/static/transcript.srt'},
+        ]))
+        self.compat.start()
+
+
+    def test_bundle_creation(self):
+
+        with mock.patch('openedx_blockstore_relay.transfer_data.requests') as mock_requests:
+            with mock.patch('openedx_blockstore_relay.transfer_data.TransferBlock.transfer_block_to_bundle'):
+                transfer_to_blockstore(self.block_key, collection_uuid=self.COLLECTION_UUID)
```

Comment:
Can we add any asserts here or do we just need to test that no exceptions occurred?

Reply:
Right now if an error happens during uploads of the assets, there is no rollback. We will have to wait for the draft/publish workflow to be able to add that. So cannot see how having this function return something will be useful. For now as long as there are no exceptions this succeeded.